src/render/utils/variants.ts

```diff
@@ -58,13 +58,39 @@ export function resolveVariantFromProps(
     currentValues: ResolvedValues = {},
     currentVelocity: ResolvedValues = {}
 ) {
+    /**
+     * If the variant definition is a function, resolve.
+     */
+    if (typeof definition === "function") {
+        definition = definition(
+            custom ?? props.custom,
+            currentValues,
+            currentVelocity
+        )
+    }
```

Comment:
As this is now in there twice, should we pull it into a `resolve` function?

Reply:
I did think about reducing duplication but if anything we'd probably just be padding out the file given that the only real duplication is the typeof check

Reply:
Ah, right, that makes sense.
Well, if you name the function right you wouldn't need the comment and there's the `??` but I see your point.